{"additions": 1, "auther_ref": "wf-573-ppc64le", "auther_sha": "d57ea911dea18738857d926861c161a884530487", "author": "smuzaffar", "body": "Workflow `573.0` for `ppc64le` IBs fails with error\r\n```\r\n  File \"Configuration/Generator/sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen_cff.py\", line 32, in <module>\r\n    \" OL_PREFIX={0} \".format(os.environ['CMS_OPENLOOPS_PREFIX']),\r\n  File \"python3/3.9.6/lib/python3.9/os.py\", line 679, in __getitem__\r\n    raise KeyError(key) from None\r\nKeyError: 'CMS_OPENLOOPS_PREFIX'\r\n```\r\nthis is because for `ppc64le` `openloops` is not build and env `CMS_OPENLOOPS_PREFIX` is not available. \r\n\r\nThe PR checks the variable before using it. Workflow `573` is still going to fail with error\r\n```\r\nMatrix_Element_Handler::BuildProcesses(): Looking for processes ME_Generators::LoadGenerator(): Try loading 'OpenLoops' from 'libSherpaOpenLoops'.\r\nLibrary_Loader::LoadLibrary(): Failed to load library 'libSherpaOpenLoops.so'.\r\nME_Generators::LoadGenerator(): ME generator 'OpenLoops' not found. Ignoring it.\r\nLibrary_Loader::LoadLibrary(): 573.0_sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+HARVESTGEN/thread124242_0/Process/Amegic/lib/libProc_P2_2.so: cannot open shared object file: No such file or directory\r\nLibrary_Loader::LoadLibrary(): Failed to load library 'libProc_P2_2.so'.\r\n```\r\nas the library `libProc_P2_2.so` ( bundled in gridpacks) is built for `x86_64` and not going to work on `ppc64le and aarch64`", "branch": "master", "changed_files": 1, "comments": 13, "commits": 2, "created_at": "1632736217", "deletions": 1, "labels": ["generators-approved", "fully-signed", "orp-pending", "tests-started", "code-checks-approved"], "milestone": "CMSSW_12_1_X", "number": 35425, "release-notes": [], "review_comments": 3, "state": "open", "title": "check CMS_OPENLOOPS_PREFIX env variable before using it", "updated_at": "1632811775", "user": "smuzaffar"}